internal/pkg/cli/task_run.go

```diff
@@ -93,12 +94,14 @@ type runTaskOpts struct {
 	sel     appEnvSelector
 	spinner progress
 
-	deployer taskDeployer
 
 	// Fields below are configured at runtime.
+	deployer             taskDeployer
 	repository           repositoryService
 	runner               taskRunner
+	deploy               func() error
```

Comment:
this is a bit odd because `configure*` fns are used to only create clients, but this one is used to actually to do a deployment.

Instead of having `deploy` as a field, which results in us not being able to test if `DeployTask` is called, what do you think of having `deployOpts []awscloudformation.StackOption` as a field?

This way we would be able to test if the ExecutionRole is passed while calling `DeployTask` if the user ran the command in an environment. Does that make sense?

Reply:
We are still able to test if `DeployTask` and whether the optional StackOptions are being passed, because we still have the field `o.deployer` which is being mocked.

With that being said! I think it's a good idea to have the `deployOpts` field because I agree that it is more readable this way and that `configure` should be used for clients (and StackOptions, in this case)